src/providers/extraction_providers/visionExtractionProvider.ts

```diff
@@ -0,0 +1,74 @@
+import { ExtractionProvider } from './extractionProvider';
+import { Image, ExtractedImage } from '../../models/data_models/image';
+import { AssignedLabel, DetectedObject, Coordinate } from '../../models/data_models/extraction';
+
+import * as vision from '@google-cloud/vision';
+
+const enum requestTypes {
+	labelDetection = 'LABEL_DETECTION',
+	objectLocalization = 'OBJECT_LOCALIZATION',
+}
+
+export class VisionExtractionProvider implements ExtractionProvider {
+	protected createAssignedLabel(
+		label: vision.protos.google.cloud.vision.v1.IEntityAnnotation
+	): AssignedLabel {
+		return {
+			description: label?.description || '',
+			confidence: label?.score || 0,
+		};
+	}
+
+	protected createBoundingPoly(
+		boundingPoly: vision.protos.google.cloud.vision.v1.INormalizedVertex
+	): Coordinate {
+		return {
+			x: boundingPoly?.x || 0,
+			y: boundingPoly?.y || 0,
+		};
+	}
+
+	protected createDetectedObject(
+		object: vision.protos.google.cloud.vision.v1.ILocalizedObjectAnnotation
+	): DetectedObject {
+		const bounding_poly: Array<Coordinate> = object.boundingPoly!.normalizedVertices!.map((coord) =>
+			this.createBoundingPoly(coord)
+		);
+		return {
+			object_name: object?.name || '',
+			confidence: object?.score || 0,
+			bounding_poly: bounding_poly,
+		};
+	}
+
+	protected async extractSingleImage(image: Image): Promise<ExtractedImage> {
+		const client = new vision.ImageAnnotatorClient();
```

Comment:
Is it necessary to create a client every time? And is there much overhead? This might be a possible spot for caching.